osu.Game.Rulesets.Catch/UI/CatchPlayfield.cs

```diff
@@ -48,12 +49,22 @@ public CatchPlayfield(BeatmapDifficulty difficulty, Func<CatchHitObject, Drawabl
                 Origin = Anchor.TopLeft,
             };
 
+            comboDisplay = new CatchComboDisplay
```

Comment:
why is this in the playfield and not part of the legacy catcher?

Reply:
`CatcherArea` limits it's height to the catcher height itself, so it's not possible to include this inside there. And to me, it makes sense to put the catch-specific combo counter in the `CatchPlayfield` itself. 

But reading this piece of code again, the only thing I missed thinking about at is how the members in here are very specific to the legacy counter instead of being set to full size and let `LegacyComboCounter` set its size properly, I'll change that.

Reply:
Even though it always tracks the catcher position? I'm not so sure about that.

Also just because its high is limited doesn't mean that's an issue unless it's masking.

Reply:
I'm making `CatchComboDisplay` an own component not tied directly to catcher because I wasn't sure lazer's combo counter implementation would be the same as legacy's tracking the catcher position.

That was one of the things I was worried about while implementing this, I could cache `CatcherArea` and let `LegacyComboCounter` do the tracking itself if that's what you would like to see removed from here.

Reply:
I think it'd be better (or in other words: less complicated) if it was in `CatcherArea` if feasible. Would save the transfer of `X` in `Update()` at the least (speaking of which, it seems to me that it can be one frame late as done currently). Being outside bounds as long as masking is off is fine with me.

Reply:
`CatcherArea` will not get rid of the X transfer though, moving it into `Catcher` will do (which would be far worse given that the catcher component is already complex enough).

Reply:
Fair enough, doing it more locally is better anyway in my book.

Reply:
I've attempted to move it into `Catcher`, had to add a 300 bottom-margin but that was fine, until I realized that `Catcher` has a logic to change the entire drawing direction of its content based on the player's desired direction, which affects the combo counter and I believe it shouldn't happen. and honestly, it sounds kinda illogical for `Catcher` to isolate its direction changing logic to its *content* only, away from the combo counter, as if the counter shouldn't be inside at all.

Anyways I've pushed a branch for this as my opinion alone is probably not enough to decide, here's a two-dots diff between current commits and my attempt of moving to `Catcher`:  https://github.com/frenzibyte/osu/compare/catch-combo-counter..combo-counter-alternate-way

Reply:
You have a point, I'm not 100% sure on `Catcher` any more looking at it... I still wouldn't say `Playfield` is good either, as it's way too general a component to have the counter in - it should be either `CatcherArea` or `Catcher`, preferably the former.

I think the ultimate solution might be to do something like

```
CatcherArea
   Container
       LegacyComboCounter
       Catcher
```

The container would be to make adjusting this:

https://github.com/ppy/osu/blob/25ee1fa7a231c2f2627193da8df957b51b51f783/osu.Game.Rulesets.Catch/UI/CatcherArea.cs#L106-L107

less painful (otherwise you'd have to transfer to both `MovableCatcher` and the counter).

Reply:
From a legacy perspective, yes it's too general to be in playfield, but from lazer's perspective which is what I envision, I think it being at the playfield makes sense as an extra HUD thing specific to the ruleset (heck I think moving it to `DrawableCatchRuleset` makes more sense for that).

But I would defer that until an actual design for osu!lazer's default skin exists and the combo counter behavior is decided (for OML skinning and otherwise), alright will move it to `CatcherArea`.